pkg/private/pkg_files.bzl

```diff
@@ -0,0 +1,321 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Internal functions for processing pkg_file* instances.
+
+Concepts and terms:
+
+  DestFile: A provider holding the source path, attributes and other
+            information about a file that should appear in the package.
+            When attributes are empty in DestFile, we let the package
+            tool decide their values.
+
+  content map: The map of destination paths to DestFile instances. Note that
+               several distinct destinations make share the same source path.
+               Attempting to insert a duplicate entry in the content map is
+               an error, because it means you are collapsing files together.
+               We may want to relax this in the future if their DestFiles
+               are equal.
+
+  manifest: The file which represents the content map. This is generated
+            by rule implementations and passed to the build_*.py helpers.
+"""
+
+load("//:path.bzl", "compute_data_path", "dest_path")
+load(
+    "//:providers.bzl",
+    "PackageArtifactInfo",
+    "PackageDirsInfo",
+    "PackageFilegroupInfo",
+    "PackageFilesInfo",
+    "PackageSymlinkInfo",
+)
+
+# Possible values for entry_type
```

Comment:
There should be a reciprocal comment pointing to `private/build_tar.py` here.

Reply:
added